bundle/edu.gemini.itc/src/main/scala/edu/gemini/itc/base/Recipe.scala

```diff
@@ -93,11 +96,20 @@ object Recipe {
     ItcCcd(singleSNRatio, totalSNRatio, r.peakPixelCount, r.instrument.wellDepth, r.instrument.gain, Warning.collectWarnings(r))
   }
 
+  // === Java helpers
+
+  // One result (CCD) and a simple set of charts, this covers most cases.
   def serviceResult(r: SpectroscopyResult, charts: java.util.List[SpcChartData]): ItcSpectroscopyResult =
-    ItcSpectroscopyResult(List(toCcdData(r, charts.toList)), charts.toList)
+    ItcSpectroscopyResult(List(toCcdData(r, charts.toList)), List(SpcChartGroup(charts.toList)))
+
+  // One result (CCD) and a set of groups of charts, this covers NIFS (1 CCD and separate groups for IFU cases).
+  def serviceGroupedResult(r: SpectroscopyResult, charts: java.util.List[java.util.List[SpcChartData]]): ItcSpectroscopyResult =
+    ItcSpectroscopyResult(List(toCcdData(r, charts.toList.flatten)), charts.toList.map(l => SpcChartGroup(l.toList)))
```

Comment:
I thought you would normally use `charts.asScala`, is the `toList` some of the automatic conversions?